File: Solutions/1695. Maximum Erasure Value/1695. Maximum Erasure Value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        int i=0,j=0,res=0,tmp=0;
        var s=new HashSet<Integer>();
        while(i<nums.length&&j<nums.length){
            if(!s.contains(nums[i])){
                s.add(nums[i]);
                tmp+=nums[i++];
                res=Math.max(res,tmp);
            }else{
                s.remove(nums[j]);
                tmp-=nums[j++];
            }
        }
        return res;
    }
}

File: Solutions/1695. Maximum Erasure Value/1695. Maximum Erasure Value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        i,j,tmp,res=0,0,0,0
        s=set()
        while i<len(nums) and j<len(nums):
            if nums[i] not in s:
                s.add(nums[i])
                tmp+=nums[i]
                i+=1
                res=max(res,tmp)
            else:
                s.remove(nums[j])
                tmp-=nums[j]
                j+=1
        return res